pkg/ccl/importccl/import_stmt.go

```diff
@@ -383,163 +384,191 @@ func importPlanHook(
 			}
 		}
 
-		var tableDescs []*sqlbase.TableDescriptor
-		var jobDesc string
-		seqVals := make(map[sqlbase.ID]int64)
-		if importStmt.Bundle {
-			store, err := storageccl.ExportStorageFromURI(ctx, files[0], p.ExecCfg().Settings)
-			if err != nil {
-				return err
-			}
-			defer store.Close()
-			raw, err := store.ReadFile(ctx, "")
-			if err != nil {
-				return err
-			}
-			defer raw.Close()
-			reader, err := decompressingReader(raw, files[0], format.Compression)
-			if err != nil {
-				return err
-			}
-			defer reader.Close()
+		var tableDetails []jobspb.ImportDetails_Table
+		jobDesc, err := importJobDescription(importStmt, nil, files, opts)
+		if err != nil {
+			return err
+		}
 
-			var match string
-			if table != nil {
-				match = table.TableName.String()
-			}
-			fks := fkHandler{skip: skipFKs, allowed: true, resolver: make(fkResolver)}
-			switch format.Format {
-			case roachpb.IOFileFormat_Mysqldump:
-				evalCtx := &p.ExtendedEvalContext().EvalContext
-				tableDescs, err = readMysqlCreateTable(ctx, reader, evalCtx, defaultCSVTableID, parentID, match, fks, seqVals)
-			case roachpb.IOFileFormat_PgDump:
-				evalCtx := &p.ExtendedEvalContext().EvalContext
-				tableDescs, err = readPostgresCreateTable(reader, evalCtx, p.ExecCfg().Settings, match, parentID, walltime, fks, int(format.PgDump.MaxRowSize))
-			default:
-				return errors.Errorf("non-bundle format %q does not support reading schemas", format.Format.String())
-			}
+		if importStmt.Into {
+			found, err := p.ResolveMutableTableDescriptor(ctx, table, true, sql.ResolveRequireTableDesc)
 			if err != nil {
 				return err
 			}
-			if tableDescs == nil && table != nil {
-				return errors.Errorf("table definition not found for %q", table.TableName.String())
-			}
 
-			descStr, err := importJobDescription(importStmt, nil, files, opts)
-			if err != nil {
+			// Take the table offline for import.
+			importing := found.TableDescriptor
+			importing.State = sqlbase.TableDescriptor_IMPORTING
```

Comment:
What does this do to the table? Users cannot read or write it during this time?

Reply:
yeah, similar to ADD state: can't lease it in SQL code so no reads or writes.